lib/NormalModule.js

```diff
@@ -284,15 +284,22 @@ class NormalModule extends Module {
 				return callback();
 			}
 
+			const isMJS = this._source._name && /\.mjs$/.test(this._source._name);
+			const sourceTypes = ["module"];
+			if(!isMJS) {
+				sourceTypes.push("script");
+			}
+
+			const source = this._source.source();
 			try {
-				this.parser.parse(this._source.source(), {
+				this.parser.parse(source, {
+					sourceTypes: sourceTypes,
```

Comment:
Passing an array as `sourceType` is not supported. These options [are not passed to acorn](https://github.com/webpack/webpack/blob/b2a6085d9684cb8bb3c47cbcb778413844d70d53/lib/Parser.js#L1167) anyway (@sokra could you clarify the purpose of this second argument `initialState`)?